src/frequenz/dispatch/_managing_actor.py

```diff
@@ -106,26 +103,45 @@ async def run():
 
     def __init__(
         self,
-        actor: Actor | Set[Actor],
-        dispatch_type: str,
         running_status_receiver: Receiver[Dispatch],
-        updates_sender: Sender[DispatchUpdate] | None = None,
+        actor: Actor | Set[Actor] | None = None,
     ) -> None:
         """Initialize the dispatch handler.
 
         Args:
-            actor: A set of actors or a single actor to manage.
-            dispatch_type: The type of dispatches to handle.
             running_status_receiver: The receiver for dispatch running status changes.
-            updates_sender: The sender for dispatch events
+            actor: Optional, but should be set later in set_actor(). A set of
+                actors or a single actor to manage.
         """
         super().__init__()
         self._dispatch_rx = running_status_receiver
-        self._actors: frozenset[Actor] = frozenset(
-            [actor] if isinstance(actor, Actor) else actor
+        self._actors: frozenset[Actor] = (
+            frozenset()
+            if actor is None
+            else frozenset([actor] if isinstance(actor, Actor) else actor)
         )
-        self._dispatch_type = dispatch_type
-        self._updates_sender = updates_sender
+        self._updates_channel = Broadcast[DispatchUpdate](
+            name="dispatch_updates_channel", resend_latest=True
+        )
+        self._updates_sender = self._updates_channel.new_sender()
+
+    def set_actor(self, actor: Actor | Set[Actor]) -> None:
+        """Set the actor to manage.
+
+        Args:
+            actor: A set of actors or a single actor to manage.
+        """
+        self._actors = (
+            frozenset([actor]) if isinstance(actor, Actor) else frozenset(actor)
+        )
+
+    def new_receiver(self) -> Receiver[DispatchUpdate]:
+        """Create a new receiver for dispatch updates.
+
+        Returns:
+            A new receiver for dispatch updates.
+        """
+        return self._updates_channel.new_receiver()
```

Comment:
Oh, yes! I also thought about this, but I didn't start to address this but was my next step. I think this actor should also stop being an actor and just be a background service.

I'm not sure it is needed with my approach, but I think it would be convenient anyway, I see no advantages in delegating the creation of the channel to the outside.